server/routes/index.ts

```diff
@@ -745,21 +747,20 @@ export function defineRoutes(router: IRouter) {
   router.delete(
     {
       path: `${API_PREFIX}/configuration/cache`,
-      validate: false,
+      validate: {
+        body: schema.object({
```

Comment:
Should the body of this request be validated if data sources are disabled?

Would datasourceId be applicable to all endpoints?

Reply:
Can we get the current datasource id from local storage and choose the correct client accordingly in the API handler and not require dataSourceId in the body of this request?

Reply:
I'm looking into seeing if its possible to use `sessionStorageFactory` to keep track of the current data source and use what's stored to determine the client to you to avoid having to pass datasourceId into all of the routes.

Reply:
Repurposed this issue: https://github.com/opensearch-project/security-dashboards-plugin/issues/1816, to decide on how to maintain the context within our plugin. I will not address this question in this PR and will track it there. Right now the two competing ways of doing so would be to lift the state/setstate one level up to be shared by all tabs, or by using session/local storage so each request wrapped with the datasource helper function can read from it instead of getting passed one by the individual tabs.